packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -177,8 +188,8 @@ module.exports = {
 
       // First, run the linter.
       // It's important to do this before Babel processes the JS.
-      {
-        test: /\.(js|jsx)$/,
+      !paths.useTSLint && {
+        test: /\.(js|jsx|tsx?)$/,
```

Comment:
Would be good to support using eslint with typescript too (if `babel-eslint` works with typescript). eslint in general has much better rules than tslint (e.g. the indent rule or the various jsx lints); it's just not type-aware.